### PR TITLE
Inject a virtual SDT instead of mapping PID 17 from the first PMT

### PR DESCRIPTION
**Pull request by Jalle19**
Should fix accidentally overwriting PID 18 (the virtual EIT) on the DDCI adapter with an SDT that has been mapped from PID 17 to PID 18.

Still not runtime tested, but the test cases pass now.

### CODE REVIEW

**Comment by Jalle19** [comment]
Considering rewriting `ddci_create_epg` to return just the table, then we can use `buffer_to_ts` just like we do for PAT, SDT and PMTs. I also believe it's not necessary to write an EIT for the virtual CA service ID (`MAKE_SID_FOR_CA`), so we could skip that. The main point is that the EIT PID should be present, that's all the standard says.

File: src/ddci.c [hunk 1]
```diff
@@ -762,12 +797,22 @@ int ddci_add_psi(ddci_device_t *d, uint8_t *dst, int len) {
...
     int psi_len;
+
+    // Add PAT
     if (ctime - d->last_pat > 500) {
         psi_len = ddci_create_pat(d, psi);
         pos += buffer_to_ts(dst + pos, len - pos, psi, psi_len, &d->pat_cc, 0);
         d->last_pat = ctime;
     }
 
+    // Add SDT
```
Comment (catalinii) [inline src/ddci.c:808]: Is best if this is after the PMT because the pmt->name can be populated based on the SDT.

**Comment by catalinii** [comment]
I added this for my CAM, I can test it next time it needs an update.

File: src/ddci.c (hunk 1 above)
Comment (Jalle19) [reply src/ddci.c:808]: The virtual SDT doesn't contain any service names at the moment, only service IDs and some status flags.

**Comment by catalinii** [comment]
I am fine either way...I will test this later anyway

**Comment by Yuri666** [comment]
Now my old Viaccess CI+ 1.2 CAM get URI and start decrypting after 3-4 secs instead 10-12 secs when without patch. Now decrypting starts at DDCI as fast as in embedded CI

**Comment by Jalle19** [comment]
Wow, that's great!

**Comment by Jalle19** [comment]
Runtime testing this too right now, so far so good

**Comment by Jalle19** [comment]
https://github.com/Jalle19/minisatip/commit/5a7c363a934470ce916942751e047ccb1a27bae2

**Comment by Jalle19** [comment]
Rebased

**Pull request merged**